first-read-last-write-cache/src/access.rs

```diff
@@ -248,4 +248,115 @@ mod tests {
             }
         )
     }
+
+    #[test]
+    fn test_err_merge_left_read_neq_right_read() {
+        let first_read = 1;
+        let mut value = create_value(first_read);
+        let mut left = Access::Read(value.clone());
+
+        let second_read = 2;
+        let mut value2 = create_value(second_read);
+
+        assert_eq!(left.merge(Access::Read(value2.clone())),
+                   Err(MergeError::ReadThenRead {
+                       left: value.clone(),
```

Comment:
No need for `clone` here (the same for other tests). Please run `cargo test check`, it will show all the warnings.